spec/defra_ruby/exporters/helpers/date_range_spec.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe DefraRuby::Exporters::Helpers::DateRange do
+  describe ".generate_date_ranges" do
+    let(:one_month) { 1 }
+    let(:two_months) { 2 }
+
+    context "when the combination of 'start' and 'end' dates and the 'range_months' results in an ambiguous range" do
+      context "and the the range includes January 1st" do
+        subject(:date_ranges) do
+          start_date = Date.new(2019, 2, 5)
+          end_date = Date.new(2019, 2, 17)
+          described_class.generate_date_ranges(start_date, end_date, two_months)
+        end
+
+        it "uses January 1st as a starting point" do
+          expect(date_ranges.count).to eq(1)
```

Comment:
More for consistency than anything else, in the subsequent higher level contexts the expectation on the size of the `date_ranges` array is brought out into its own test, which includes confirming its of type `Array`.

I'd prefer to do the same within these contexts as well.

Reply:
We've discussed why I don't want to change this. I've added comments to the spec in 63c3895 explaining this.